website/docs/r/virtual_machine_configuration_policy_assignment.html.markdown

```diff
@@ -0,0 +1,140 @@
+---
+
+# azurerm_virtual_machine_configuration_policy_assignment
+
+Manages a Virtual Machine Configuration Policy Assignment.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-gca"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "example-vnet"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_network_interface" "example" {
+  name                = "example-nic"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.example.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_windows_virtual_machine" "example" {
+  name                = "examplevm"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  size                = "Standard_F2"
+  admin_username      = "adminuser"
+  admin_password      = "P@$$w0rd1234!"
+  network_interface_ids = [
+    azurerm_network_interface.example.id,
+  ]
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2016-Datacenter"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_virtual_machine_configuration_policy_assignment" "example" {
+  name               = "example-gca"
+  location           = azurerm_linux_virtual_machine.example.id
+  virtual_machine_id = azurerm_linux_virtual_machine.example.id
+  guest_configuration {
+    name    = "WhitelistedApplication"
+    version = "1.*"
+
+    parameter {
+      name  = "[InstalledApplication]bwhitelistedapp;Name"
+      value = "NotePad,sql"
+    }
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Virtual Machine Configuration Policy Assignment. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure location where the Virtual Machine Configuration Policy Assignment should exist. Changing this forces a new resource to be created.
+
+* `virtual_machine_id` - (Required) The resource ID of the Virtual Machine which this Guest Configuration Assignment should apply to. Changing this forces a new resource to be created.
+
+---
+
+* `guest_configuration` - (Required)  A `guest_configuration` block as defined below.
```

Comment:
this doesn't match schema.

given policy is in the name, and policy_assingment.policy is an inline policy not a policy id that is assigned, maybe policy doesn't make the most sens

configuration_checks? checks?
```
configuration {
  parameter {
    name = "someprop"
    matches = "somevalue"
  }
}
```

WDYT? the schema and names here are confusing and all over and i'm not sure what the best UX here would be

Reply:
You are absolutely correct! I forgot to change this after I changed the schema. I just fixed this.